website/docs/cli/cloud/settings.mdx

```diff
@@ -91,6 +91,8 @@ Alternatively, the `cloud` block can be configured with the following environmen
 
 - `TF_HOSTNAME` - The hostname of a Terraform Enterprise installation. Serves as a fallback if `hostname` is not specified in the cloud configuration. If both are specified, the configuration takes precendence. 
 
+- `TF_WORKSPACE` - The workspace to select from the available set. The `cloud` block interprets this variable as the `name` value of the `workspaces` attribute allowing you to omit this attribute completely. It does not alter the existing behavior of how Terraform interprets this variable. [Read more about TF_WORKSPACE](https://www.terraform.io/cli/config/environment-variables#tf_workspace)
```

Comment:
This is cool! Two questions for you:
1. Should we now mark "workspaces" argument as optional? Should organization also be marked similarly? And should we say that if those things aren't set, users WILL. be expected to supply that on the command line?
2. What happens if you have the arguments configured in the configuration and then try to supply these values also with command line variables? Do the command line variables override what's in the config?
3. The PR comments bring up something interesting - the idea that you could theoretically leave the cloud block empty in your config and interactively set it via these command line variables. I'm wondering whether we should add something to the docs telling users this and also recommending when it is and isn't appropriate to do that.

Reply:
1. I'm not sure about the first point. The argument is still technically required, we're just adding to the ways that data can be set.
2. The configuration file takes precedence over environment variables.
3. For a CI environment, I think the ideal case would be that the cloud configuration block be empty

Here's the [RFC](https://docs.google.com/document/d/1UHVz-_Xkom4RrcW4540Y_zKZpTs92oj0ZejIhHb-K3w) that covers the configuration environment variables. There is also an RFC that covers setting credentials with an environment variable, but that work has not yet started.

Reply:
Great questions!

On your first question, I have thought about this and my answer depends on which context we use to define the word `required`. The attribute (both `organization` and `workspaces`) is _required_ in the sense that the value must either exist in the configuration or as an environment variable. However it is not required to be explicitly defined in your configuration as it can be set as an environment variable. I think however that marking it as `optional` will indicate this value can be omitted completely, which is not true... the value must in fact still exist. 

On your second question, the configuration always takes precedence over the environment so you can't override your configuration with env variables. The environment variables behave more like a fallback option to make it easier for customers who want to configure Terraform non-interactively. From my understanding however, is that explicitly configuring Terraform in your configuration is still the recommended approach. 

Ditto on your third point! Always helpful to provide extra guidance with this feature.

Reply:
Perhaps I did not make in clear in the docs, but the cloud configuration will only look for `TF_WORKSPACE` if the `workspaces` block is completely omitted, so:

```hcl
terraform {
  cloud {
     workspaces {}
   }
}
```
is invalid even if the env variable is set and will yield an error.